subgraphs/aave-forks/src/mapping.ts

```diff
@@ -372,707 +513,584 @@ export function _handleReserveDataUpdated(
   liquidate.save();
+}
 
-  protocol.cumulativeLiquidateUSD = protocol.cumulativeLiquidateUSD.plus(
-    liquidate.amountUSD
-  );
-  protocol.save();
-  market.cumulativeLiquidateUSD = market.cumulativeLiquidateUSD.plus(
-    liquidate.amountUSD
+export function _handleFlashLoan(
+  asset: Address,
+  amount: BigInt,
+  account: Address,
+  procotolData: ProtocolData,
+  event: ethereum.Event,
+  premiumAmount: BigInt,
+  flashloanPremium: _FlashLoanPremium
+): void {
+  const market = getMarketFromToken(asset, procotolData);
+  if (!market) {
+    log.warning("[_handleFlashLoan] market for token {} not found", [
+      asset.toHexString(),
+    ]);
+    return;
+  }
+  const manager = new DataManager(
+    market.id,
+    market.inputToken,
+    event,
+    procotolData
   );
-  market.save();
+  const tokenManager = new TokenManager(asset, event);
+  const amountUSD = tokenManager.getAmountUSD(amount);
+  const premiumUSDTotal = tokenManager.getAmountUSD(premiumAmount);
+  const flashloan = manager.createFlashloan(asset, account, amount, amountUSD);
+  flashloan.feeAmount = premiumAmount;
+  flashloan.feeAmountUSD = premiumUSDTotal;
+  flashloan.save();
+
+  let reserveFactor = market.reserveFactor;
+  if (!reserveFactor) {
+    reserveFactor = BIGDECIMAL_ZERO;
+  }
+  const protocolRevenueShare = reserveFactor;
+  let premiumUSDToProtocol = premiumUSDTotal.times(protocolRevenueShare);
+  let premiumUSDToLP = premiumUSDTotal.minus(premiumUSDToProtocol);
+  let premiumUSDToDeduct = premiumUSDTotal;
+  const premiumRateTotal = flashloanPremium.premiumRateTotal;
+  let premiumRateToProtocol = premiumRateTotal.times(protocolRevenueShare);
+  let premiumRateToLP = premiumRateTotal.minus(premiumRateToProtocol);
+
+  if (flashloanPremium.premiumRateToProtocol.gt(BIGDECIMAL_ZERO)) {
+    // according to https://github.com/aave/aave-v3-core/blob/29ff9b9f89af7cd8255231bc5faf26c3ce0fb7ce/contracts/interfaces/IPool.sol#L634
+    // premiumRateToProtocol is the percentage of premium to protocol
+    premiumUSDToProtocol = premiumUSDTotal
+      .times(flashloanPremium.premiumRateToProtocol)
+      .plus(premiumUSDToProtocol);
+    premiumRateToProtocol = premiumRateTotal
+      .times(flashloanPremium.premiumRateToProtocol)
+      .plus(premiumRateToProtocol);
+
+    // premium to LP
+    premiumUSDToLP = premiumUSDTotal.minus(premiumUSDToProtocol);
+    premiumRateToLP = premiumRateTotal.minus(premiumRateToProtocol);
+    // this part of the premium is transferred to the treasury and not
+    // accrued to liquidityIndex and thus no need to deduct
+    premiumUSDToDeduct = premiumUSDToDeduct.minus(
+      premiumUSDTotal.times(flashloanPremium.premiumRateToProtocol)
+    );
+  }
 
-  // update usage metrics
-  snapshotUsage(
-    protocol,
-    event.block.number,
-    event.block.timestamp,
-    liquidate.liquidatee,
-    EventType.LIQUIDATEE,
-    true // only count this liquidate as new tx
+  const feeToProtocol = manager.getOrUpdateFee(
```

Comment:
isnt this updated in the event that updates this?

Reply:
not sure I understand what you mean here.